src/server/game/Spells/Spell.cpp

```diff
@@ -2676,6 +2677,11 @@ SpellMissInfo Spell::DoSpellHitOnUnit(Unit* unit, uint32 effectMask, bool scaleA
         if (effectMask & (1 << effectNumber))
             HandleEffects(unit, NULL, NULL, effectNumber, SPELL_EFFECT_HANDLE_HIT_TARGET);
 
+    // Check for SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER and eventually cast the interrupt spell if the target is a non-player
+    if (m_spellInfo->HasAttribute(SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER))
+        if (unit->GetTypeId() != TYPEID_PLAYER)
```

Comment:
That is how C++ already works. Please don't comment if you have no idea what you are talking about.